### ISSUE TITLE
Combine Two Roberta Classifier for two input text column and come up with single outcome class label

### ISSUE THREAD

**Issue by suryapa1** [ThilinaRajapakse/simpletransformers#553]
**Describe the bug**
I have two text columns as short_text, description in my dataset, I would like to have two Roberta models feed through these two input COLUMNS as mentioned in below links and would like to combine the results as shown in the attached diagram.

Is it possible to tweak existing simple transformers based on variants mentioned in the link (towards data science) as an example
1. Dual BERT
2.SIAMESE BERT

Any sample code snippets are greatly appreciated.

**REFERENCE LINKS TO Reproduce**
https://towardsdatascience.com/siamese-and-dual-bert-for-multi-text-classification-c6552d435533

**Additional context**
<img width="735" alt="Screenshot 2020-07-06 at 5 31 22 PM" src="https://user-images.githubusercontent.com/6042186/86591258-d8c18f80-bfae-11ea-8362-d444f1683ca6.png">

**Comment by ThilinaRajapakse** [comment]
Just off the top of my head, you will have to handle two things.

1. You will need to create a model class for Dual/Siamese BERT (each). You could look at [how](https://github.com/ThilinaRajapakse/simpletransformers/blob/aa8e6a6337938aa9b8e5e8c8427b1e5b90aed8ef/simpletransformers/classification/transformer_models/bert_model.py#L7) the `BertForSequenceClassification` is implemented for guidance. 

```python
    def __init__(self, config, weight=None):
        super(BertForSequenceClassification, self).__init__(config)
        self.num_labels = config.num_labels

        self.bert = BertModel(config)
        self.dropout = nn.Dropout(config.hidden_dropout_prob)
        self.classifier = nn.Linear(config.hidden_size, self.config.num_labels)
        self.weight = weight

        self.init_weights()
```

Note how this implementation has a base BERT model and then the linear classifier layer. The base model is essentially the TRANSFORMER block + Pooling layer in your diagram.

This means you can do something like:

```python
    def __init__(self, config, weight=None):
        super(BertForSequenceClassification, self).__init__(config)
        self.num_labels = config.num_labels

        self.bert_1 = BertModel(config)
        self.bert_2 = BertModel(config)
        self.dropout = nn.Dropout(config.hidden_dropout_prob)
        self.classifier = nn.Linear(config.hidden_size * 2, self.config.num_labels)
        self.weight = weight

        self.init_weights()
```

Note that I've initialized two `BertModel` instances and doubled the width of the linear classifier to account for both models.

Then, you have to modify the `forward()` method as needed. The example code given below is not tested and is unlikely to work out of the box, but hopefully, it illustrates the general idea.

```python
    def forward(
        self,
        input_ids_1=None,
        attention_mask_1=None,
        token_type_ids_1=None,
        position_ids_1=None,
        head_mask_1=None,
        input_ids_2=None,
        attention_mask_2=None,
        token_type_ids_2=None,
        position_ids_2=None,
        head_mask_2=None,
        inputs_embeds=None,
        labels=None,
    ):

        outputs_1 = self.bert_1(
            input_ids_1,
            attention_mask=attention_mask_1,
            token_type_ids=token_type_ids,_1
            position_ids=position_ids_1,
            head_mask=head_mask_1,
        )
        pooled_output_1 = outputs_1[1]

        pooled_output_1 = self.dropout(pooled_output_1)

        outputs_2 = self.bert_2(
            input_ids_2,
            attention_mask=attention_mask_2,
            token_type_ids=token_type_ids,_2,
            position_ids=position_ids_2,
            head_mask=head_mask_2,
        )
        pooled_output_2 = outputs_2[1]

        pooled_output_2 = self.dropout(pooled_output_2)

        logits = self.classifier(pooled_output_1 + pooled_output_2)

        outputs = (logits,) + outputs_1[2:] + outputs_2[2:]  # add hidden states and attention if they are here

        if labels is not None:
            if self.num_labels == 1:
                #  We are doing regression
                loss_fct = MSELoss()
                loss = loss_fct(logits.view(-1), labels.view(-1))
            else:
                loss_fct = CrossEntropyLoss(weight=self.weight)
                loss = loss_fct(logits.view(-1, self.num_labels), labels.view(-1))
            outputs = (loss,) + outputs

        return outputs  # (loss), logits, (hidden_states), (attentions)
```

2. The other thing you need to handle is loading and tokenizing the input data. For that, you can look at the `load_and_cache_data()` method implementation. You will need to make sure that the data is tokenized properly and passed to the model in the correct format.

**Comment by suryapa1** [comment]
Thanks alot for your inputs, Will take further from this.
